src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchService.kt

```diff
@@ -94,50 +114,66 @@ class MatchService(
 
 fun BoolQueryBuilder.mustWhenPresent(query: String, value: String?): BoolQueryBuilder {
   value.takeIf { !it.isNullOrBlank() }?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
+  }
+  return this
+}
+
+fun BoolQueryBuilder.shouldWhenPresent(query: String, value: String?): BoolQueryBuilder {
+  value.takeIf { !it.isNullOrBlank() }?.let {
+    this.should(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.mustWhenPresent(query: String, value: LocalDate?): BoolQueryBuilder {
   value?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
+  }
+  return this
+}
+
+fun BoolQueryBuilder.shouldWhenPresent(query: String, value: LocalDate?): BoolQueryBuilder {
+  value?.let {
+    this.should(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.must(query: String, value: Any): BoolQueryBuilder {
-  this.must(QueryBuilders.matchQuery(query, value))
+  this.must(matchQuery(query, value))
   return this
 }
 
 fun BoolQueryBuilder.mustWhenTrue(predicate: () -> Boolean, query: String, value: String): BoolQueryBuilder {
   value.takeIf { predicate() }?.let {
-    this.must(QueryBuilders
-        .matchQuery(query, it))
+    this.must(matchQuery(query, it))
   }
   return this
 }
 
 fun BoolQueryBuilder.mustMultiMatchKeyword(value: String, vararg query: String): BoolQueryBuilder {
   this.must().add(multiMatchQuery(value, *query)
-      .analyzer("whitespace")
+      .analyzer("keyword")
   )
   return this
 }
 
+fun BoolQueryBuilder.mustKeyword(value: String, query: String): BoolQueryBuilder {
+  return this.must(matchQuery(query, value).analyzer("keyword"))
+}
 
-sealed class Result<out T, out M> {
```

Comment:
Decided didn't need generics for this given it is specific to searching, makes the code a bit more readable